solidity/test/FullyBackedBondingTest.js

```diff
@@ -204,6 +204,54 @@ describe("FullyBackedBonding", function () {
     })
   })
 
+  describe("topUp", async () => {
+    const value = new BN(123)
+
+    let initialDeposit
+
+    before(async () => {
```

Comment:
Should this one be `before` or `beforeEach`? I think `before` does not work the way we expect it to work in this context.

Reply:
We defined `beforeEach` with snapshots at the top level, these all doesn't work well together. Let me figure it out.